### ISSUE TITLE
Using ConcatDataset in validation set in new mmdet v3

### ISSUE THREAD

**Issue by levan92** [open-mmlab/mmdetection#8785]
For the latest release mmdet v3, if `ConcatDataset` is used in validation dataset, how can `val_evaluator` be set in the config? 

I've tried not providing ann_file to `val_evaluator` as well as giving a list of `CocoMetric` dicts, each corresponding to the ann_file in `ConcatDataset`-- both does not work. 

Please advise. Thank you!

**Comment by levan92** [comment]
Hi, any updates on this?  @ZwwWayne  @hhaAndroid 

Could you let us know if evaluation on ConcatDataset is supported? If not, where in the code base will be a good place to modify it for support. 

Thank you!

**Comment by levan92** [comment]
Did a bit of digging. It seems like previously, the `evaluate` function is defined by the `dataset` class, and therefore, `ConcatDataset`'s `evaluate` method is able to handle separate evals for its datasets. However, now it is all subsumed under `CocoMetric` evaluator. I'm guessing we will have to create a metric/evaluator class for concat datasets?

**Comment by levan92** [comment]
Hi @hhaAndroid, any update? I can look into this if it isn't something you all are fixing at this moment. Thank you!

**Comment by levan92** [comment]
@ZwwWayne @hhaAndroid, any updates on this?

**Comment by XinyueZ** [comment]
I find out that the  it only worked for train (set).
It not worked for val and test set.
